app/controllers/post_controller.rb

```diff
@@ -0,0 +1,49 @@
+class PostsController < ApplicationController
+  before_action :find_post, only: %i[show edit update destroy]
+
+  def index
+    @posts = Post.all
+  end
+
+  def show
+    @recent_comments = @post.recent_comments
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    if @post.save
+      redirect_to @post, notice: 'Post was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @post.update(post_params)
+      redirect_to @post, notice: 'Post was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @post.destroy
+    redirect_to posts_url, notice: 'Post was successfully destroyed.'
+  end
+
```

Comment:
- [ ] Kindly remove these controller actions. We only need the index and show actions.